diff --git a/node_modules/react-native/scripts/cocoapods/helpers.rb b/node_modules/react-native/scripts/cocoapods/helpers.rb
index cdb1e14..b7370fb 100644
--- a/node_modules/react-native/scripts/cocoapods/helpers.rb
+++ b/node_modules/react-native/scripts/cocoapods/helpers.rb
@@ -3,6 +3,8 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+require 'shellwords'
+
 # Helper object to wrap the invocation of sysctl
 # This makes it easier to mock the behaviour in tests
 class SysctlChecker
@@ -31,7 +33,7 @@ class Finder
     def self.find_codegen_file(path)
         js_files = '-name "Native*.js" -or -name "*NativeComponent.js"'
         ts_files = '-name "Native*.ts" -or -name "*NativeComponent.ts"'
-        return `find #{path} -type f \\( #{js_files} -or #{ts_files} \\)`.split("\n").sort()
+        return `find #{Shellwords.escape(path)} -type f \\( #{js_files} -or #{ts_files} \\)`.split("\n").sort()
     end
 end
 
diff --git a/node_modules/react-native/scripts/cocoapods/new_architecture.rb b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
index 7d8f8e6..79bfd8e 100644
--- a/node_modules/react-native/scripts/cocoapods/new_architecture.rb
+++ b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
@@ -166,7 +166,7 @@ class NewArchitectureHelper
         excluded_info_plist = ["/Pods", "Tests", "metainternal", ".bundle", "build/", "DerivedData/"]
         projectPaths.each do |projectPath|
             projectFolderPath = File.dirname(projectPath)
-            infoPlistFiles = `find #{projectFolderPath} -name "Info.plist"`
+            infoPlistFiles = `find #{Shellwords.escape(projectFolderPath)} -name "Info.plist"`
             infoPlistFiles = infoPlistFiles.split("\n").map { |f| f.strip }
 
             infoPlistFiles.each do |infoPlistFile|
